File: java/src/main/java/finance/clarian/sdk/CashInService.java
package finance.clarian.sdk;

import java.util.Map;

/** PIX cash-in: generate and fetch dynamic charges. */
public final class CashInService {

    private final ClarianClient client;

    CashInService(ClarianClient client) {
        this.client = client;
    }

    /**
     * POST /cash-in/pix. {@code idempotencyKey} is required by the API —
     * retries with the same key return the original order.
     */
    public Map<String, Object> create(Map<String, ?> params, String idempotencyKey) {
        return client.request("POST", "/cash-in/pix", params, idempotencyKey);
    }

    /** GET /cash-in/{id}. */
    public Map<String, Object> retrieve(String orderId) {
        return client.request("GET", "/cash-in/" + Paths.escape(orderId), null, null);
    }
}

File: dotnet/Clarian.Sdk/CashInService.cs
namespace Clarian;

/// <summary>PIX cash-in: generate and fetch dynamic charges.</summary>
public sealed class CashInService
{
    private readonly ClarianClient _client;

    internal CashInService(ClarianClient client) => _client = client;

    /// <summary>
    /// POST /cash-in/pix. idempotencyKey is required by the API —
    /// retries with the same key return the original order.
    /// </summary>
    public Task<CashInOrderResponse> CreateAsync(
        CashInRequest request,
        string idempotencyKey,
        CancellationToken cancellationToken = default) =>
        _client.RequestAsync<CashInOrderResponse>(
            HttpMethod.Post,
            "/cash-in/pix",
            request,
            idempotencyKey,
            cancellationToken);

    /// <summary>GET /cash-in/{id}.</summary>
    public Task<CashInOrderResponse> RetrieveAsync(
        string orderId,
        CancellationToken cancellationToken = default) =>
        _client.RequestAsync<CashInOrderResponse>(
            HttpMethod.Get,
            "/cash-in/" + ClarianClient.PathEscape(orderId),
            null,
            null,
            cancellationToken);
}
